tests/test_unittest/test_caching.py

```diff
@@ -0,0 +1,204 @@
+import pytest
+
+import gt4py
+from gt4py.gtscript import computation, interval, PARALLEL, Field
+from gt4py.stencil_builder import StencilBuilder
+
+
+# type ignores in stencils are because mypy does not yet
+# deal with gtscript types well
+
+
+def simple_stencil(field: Field[float]):  # type: ignore
+    with computation(PARALLEL), interval(...):  # type: ignore
+        field += 1  # type: ignore
+
+
+def simple_stencil_same(field: Field[float]):  # type: ignore
+    with computation(PARALLEL), interval(...):  # type: ignore
+        field += 1  # type: ignore
+
+
+def simple_stencil_with_doc(field: Field[float]):  # type: ignore
+    """Increment the in/out field by one."""
+    with computation(PARALLEL), interval(...):  # type: ignore
+        field += 1  # type: ignore
+
+
+@pytest.fixture
+def builder():
+    """Preconfigure builder so only definition has to be added."""
+
+    def make_builder(definition, backend_name="debug", module=None):
+        """Make a builder instance a definition and default params."""
+        return StencilBuilder(
+            definition,
+            backend=gt4py.backend.from_name(backend_name),
+            options=gt4py.definitions.BuildOptions(name="foo", module=module or f"{__name__}"),
+        )
+
+    return make_builder
+
+
+def test_jit_properties(builder):
+    """
+    Test properties of :py:class:`gt4py.backend.caching.JITCachingStrategy`.
```

Comment:
Is this kind of test setup with extensive documentation at the top an established methodology? My personal feeling would be to put these comments inline (if necessary at all).

Reply:
I personally add very minimal documentation to the tests and avoid running `mypy` on them because it might end up being a lot of work and don't see clear benefits, but it might be worth a discussion in the team to decide it.

Reply:
I put comments like these because I consider tests = documentation. If you look at some of the older tests in gt4py it will take you 3 days min to figure out what behaviour they are supposed to ensure if you come in cold. If tests are well documented it becomes easier to decide for someone else down the line.

This is not an "industry standard" or "best practice", only me trying to avoid obscure tests slowing down development later (a common problem with scientific projects).
If you prefer to document detailed behaviour of the implementation somewhere else, we can move that stuff.

Mypy is not running on the tests as far as I know.

* what the actual intended behaviour even *is*
* how to adapt the test to intended changes in behaviour
* what behaviour was broken if the test fails

Reply:
After additional discussion we decided to pack the information in the docstrings into
* function names where appropriate
* inline comments where that makes more sense
in order to achieve high readability as well as higher likelihood of information being kept up to date.